### DISCUSSION TITLE
Dynamic Sleep not working  ?

### DISCUSSION THREAD

**Discussion by smcgann99** [arendst/Tasmota#17772] [Q&A] [answered]
I have a bulb running 9.1.0(lite) with SO60 set to 0 and sleep set to 50. When the light is on the load is 99% and sleep is 10.
When off its 19% and sleep goes back to 50.

But other bulbs running newer firmware such as 12.3.1(lite) go to 99% with sleep 10 when turned and stay at that even after they have been turned off.

The only way to get it back to sleep 50 is to set it again in console i.e. sleep 50, after that its OK until the light is turned on again.

Seems like a bug ?

**Comment by barbudor** [comment, marked as answer]
@s-hadinger Could it be a problem introduced by https://github.com/arendst/Tasmota/commit/b2e9001ca29f6f9b68d74debea15c6d086ca79db ?

**Reply by smcgann99** [reply]
Is that not only on ESP32 builds ?  - These are ESP8266 based.

**Reply by barbudor** [reply]
The part in `xdrv_04_light.ino:LightAnimate()` is not ESP32 dependant and seems to be the only recent change around the sleep parameter
This was introduced in dev 1.2.0.2
So you can try release 12.2.0 as it should be the latest release without those changes.
If confirmed we could then investigate further around that

**Reply by smcgann99** [reply]
It seems fine in 12.2.0(tasmota)

11:42:37.067 MQT: tele/RGB-Plug01/STATE = "Dynamic","Sleep":50,"LoadAvg":19
11:43:05.423 MQT: stat/RGB-Plug01/RESULT = {"POWER2":"ON"}
11:43:07.073 MQT: tele/RGB-Plug01/STATE = "Dynamic","Sleep":10,"LoadAvg":35
11:43:17.069 MQT: tele/RGB-Plug01/STATE = "Dynamic","Sleep":10,"LoadAvg":83
11:43:27.066 MQT: tele/RGB-Plug01/STATE = "Dynamic","Sleep":10,"LoadAvg":99,
11:43:43.635 MQT: stat/RGB-Plug01/RESULT = {"POWER2":"OFF"}
11:43:47.106 MQT: tele/RGB-Plug01/STATE = :"Dynamic","Sleep":50,"LoadAvg":36
11:43:57.071 MQT: tele/RGB-Plug01/STATE = "Dynamic","Sleep":50,"LoadAvg":19

**Reply by barbudor** [reply]
Thanks for the confirmation
I will investigate more tonight

**Reply by barbudor** [reply]
@smcgann99 I think I found a fix but I need s-hadinger to double check possible impact with ArtNet so this if for now on hold.

**Reply by smcgann99** [reply]
OK - I can always downgrade then.

**Reply by barbudor** [reply]
It has been merge now https://github.com/arendst/Tasmota/pull/17786
So the latest development binaries on http://ota.tasmota.com/tasmota/ includes that fix now

**Reply by smcgann99** [reply]
Thanks

**Reply by Einstein67** [reply]
The fix works perfect for lights!

But if there is a Buzzer configured, Sleepmode stays on Active-Sleep value, and does not switch back to Dynamic-Sleep Value. 

Edit: "SetOption111 1" and "restart 1" fixed the problem. Sorry!

**Reply by barbudor** [reply]
It looks like the problem may actually be there too in Buzzer but from much more longer than s-hadinger PR on ArtNet
I will see if I have time to check that further
